### PR TITLE
Fix handling of generic grain type codes

### PR DESCRIPTION
**Pull request by sergeybykov**
This is to fix the issue exposed by the GenericGrainTests_UseGenericFactoryInsideGrain test.

### CODE REVIEW

File: src/ClientGenerator/CSharpCodeGenerator.cs [hunk 1]
```diff
@@ -394,13 +394,13 @@ protected override void AddGetGrainMethods(GrainInterfaceData iface, CodeTypeDec
                 add(@"
                         public static {0} GetGrain(long primaryKey, string keyExt)
                         {{
-                            return Cast(global::Orleans.CodeGeneration.GrainFactoryBase.MakeKeyExtendedGrainReferenceInternal(typeof({0}), {1}, primaryKey, keyExt));
+                            return Cast(global::Orleans.CodeGeneration.GrainFactoryBase.MakeKeyExtendedGrainReferenceInternal(typeof({0}), primaryKey, keyExt));
```
Comment (gabikliot) [inline src/ClientGenerator/CSharpCodeGenerator.cs]: you removed {1}, what about the extra args now?

File: src/ClientGenerator/CSharpCodeGenerator.cs [hunk 2]
```diff
@@ -394,13 +394,13 @@ protected override void AddGetGrainMethods(GrainInterfaceData iface, CodeTypeDec
...
                         }}");
 
                 add(@"
                         public static {0} GetGrain(long primaryKey, string keyExt, string grainClassNamePrefix)
                         {{
-                            return Cast(global::Orleans.CodeGeneration.GrainFactoryBase.MakeKeyExtendedGrainReferenceInternal(typeof({0}), {1}, primaryKey, keyExt, grainClassNamePrefix));
+                            return Cast(global::Orleans.CodeGeneration.GrainFactoryBase.MakeKeyExtendedGrainReferenceInternal(typeof({0}), primaryKey, keyExt, grainClassNamePrefix));
```
Comment (gabikliot) [inline src/ClientGenerator/CSharpCodeGenerator.cs]: same question about extra arg.

**Comment by gabikliot** [comment]
Can you please explain the fix? It looks like you only removed some extra args. How that fixes it?

File: src/ClientGenerator/CSharpCodeGenerator.cs (hunk 1 above)
Comment (sergeybykov) [inline src/ClientGenerator/CSharpCodeGenerator.cs]: What about them? I believe they are still needed.

**Comment by sergeybykov** [comment]
I removed the overload of MakeGrainReferenceInternal that was taking interfaceId https://github.com/dotnet/orleans/pull/479/files#diff-edbaf55b6b490cc1ecdd45593fcf7a22L175 and all usages of it. That was a code path producing different type codes than the one that takes just interface type.

File: src/ClientGenerator/CSharpCodeGenerator.cs (hunk 1 above)
Comment (gabikliot) [inline src/ClientGenerator/CSharpCodeGenerator.cs]: Ohh, sorry, I misread the code.  The {1} applies to the removed CodeSnippetTypeMember interfaceId.
Reviewing code via GitHub is just so bad.

File: src/ClientGenerator/CSharpCodeGenerator.cs [hunk 3]
```diff
@@ -379,7 +379,7 @@ protected override void AddGetGrainMethods(GrainInterfaceData iface, CodeTypeDec
             RecordReferencedNamespaceAndAssembly(iface.Type);
             var interfaceId = GrainInterfaceData.GetGrainInterfaceId(iface.Type);
```
Comment (gabikliot) [inline src/ClientGenerator/CSharpCodeGenerator.cs]: looks like this variable interfaceId  is not used any more now. Can be removed?

**Comment by sergeybykov** [comment]
I added another test case that shows that my fix didn't actually fully solve the problem. Closing this PR for now.

**Comment by sergeybykov** [comment]
I think I now fixed it for real. Reopening.

**Pull request merged**